bluesky/plans.py

```diff
@@ -1985,10 +2026,14 @@ def scan_nd(detectors, cycler, *, per_step=None, md=None):
     """
     if md is None:
         md = {}
-    md.update({'detectors': [det.name for det in detectors],
-               'motors': [motor.name for motor in cycler.keys]})
-    md['plan_args'] = {'detectors': list(map(repr, detectors)),
-                       'cycler': repr(cycler)}
+    md = ChainMap(
+        md,
+        {'detectors': [det.name for det in detectors],
```

Comment:
I blame it on lack of sleep. I get it now: top-level metadata gets the name, plan_args gets the repr